pkg/kubectl/resource/builder.go

```diff
@@ -757,3 +774,17 @@ func HasNames(args []string) (bool, error) {
 	}
 	return hasCombinedTypes || len(args) > 1, nil
 }
+
+func (b *Builder) validateNamespace(c RESTClient, namespace string) error {
+	if !b.validateNamespaces {
+		return nil
+	}
+	if namespace == api.NamespaceDefault || namespace == api.NamespaceAll {
+		return nil
+	}
+	err := c.Get().AbsPath("api", "v1", "namespaces", namespace).Do().Error()
```

Comment:
I'm assuming that the server side is already doing the lookup to deny you
access - if you did this on the failure path on the server you'd improve
both sophisticated clients (kubectl) and naive ones (web dashboards, curl
integrations).

https://github.com/kubernetes/kubernetes/pull/28120 is an example of that -
an object that was deleted prior to creation.  I don't know that we can
easily generalize this on the server, but if we could it would mean every
client would get the benefit in a structured fashion.

On Mon, Aug 29, 2016 at 11:00 PM, Brendan Burns <notifications@github.com>
wrote:

> In pkg/kubectl/resource/builder.go
> <https://github.com/kubernetes/kubernetes/pull/31594#discussion_r76723967>
> :
>
> > @@ -757,3 +774,17 @@ func HasNames(args []string) (bool, error) {
> >  	}
> >  	return hasCombinedTypes || len(args) > 1, nil
> >  }
> > +
> > +func (b *Builder) validateNamespace(c RESTClient, namespace string) error {
> > +	if !b.validateNamespaces {
> > +		return nil
> > +	}
> > +	if namespace == api.NamespaceDefault || namespace == api.NamespaceAll {
> > +		return nil
> > +	}
> > +	err := c.Get().AbsPath("api", "v1", "namespaces", namespace).Do().Error()
>
> @smarterclayton <https://github.com/smarterclayton> I don't think we want
> to do this server side because it implies an extra lookup for every API
> request, which is a cost that I don't think we want to pay just for better
> UX.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31594/files/56a7f301ef7962e06f6f5c7a99e8c39ff219f3fa#r76723967>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p16dyHKHgEaIe6L0Hhg4Pb0VcUItks5qk5y0gaJpZM4JvJnq>
> .
>